docs/data/material/components/text-fields/text-fields.md

```diff
@@ -224,20 +224,16 @@ Make sure that the input is larger than the label to display correctly.
 
 ### type="number"
 
-Inputs of type="number" have potential usability issues:
+We do not recommend using type="number" with a Text Field due to potential usability issues:
 
 - Allowing certain non-numeric characters ('e', '+', '-', '.') and silently discarding others
 - The functionality of scrolling to increment/decrement the number can cause accidental and hard-to-notice changes
 
 and more - see [this article](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/) by the GOV.UK Design System team for a more detailed explanation.
 
-For number validation, one viable alternative is to use the default input type="text" with the _pattern_ attribute, for example:
+For number validation, you can try Base UI's [Number Input](/base-ui/react-number-input/).
 
-```jsx
-<TextField inputProps={{ inputMode: 'numeric', pattern: '[0-9]*' }} />
-```
-
-In the future, we might provide a [number input component](https://github.com/mui/material-ui/issues/19154).
+You can also upvote [this GitHub issue](https://github.com/mui/material-ui/issues/19154) to see it arrive in Material UI sooner!
```

Comment:
We often say "sooner" in this context but as a reader it just makes me wonder, "sooner than what?" Instead, maybe we should frame it terms of expressing your need for it. I dunno, I guess that's nitpicky of me 😅

```suggestion
You can upvote [this GitHub issue](https://github.com/mui/material-ui/issues/19154) to express your need for a Number Input component in Material UI.
```

Reply:
Though in this case the issue already has enough upvotes at this point, and we're almost for sure going to do it, how about something like:

"You can follow (or "subscribe to?") this GitHub issue to follow the progress of introducing a Number Input component to Material UI" @samuelsycamore

Reply:
Cool yeah that makes sense, I'd say go with "follow this GitHub issue" 👍

Reply:
Updated!

Reply:
Good work @mj12albert Thank you ^^

Reply:
why is this not a major priority?  Surely entering numbers a completely fundamental requirement?